### ISSUE TITLE
fortios_router_ospf not removed on destroy

### ISSUE THREAD

**Issue by poroping** [fortinetdev/terraform-provider-fortios#129]
Resource does not seem to attempt to destroy. This is causing issues as the OSPF config can reference interfaces which then cannot be destroyed.

Resource applied:
```
resource "fortios_router_ospf" "ospf" {
  default_information_metric      = 10
  default_information_metric_type = "2"
  default_information_originate   = "enable"
  default_metric                  = 10
  distance                        = 110
  distance_external               = 110
  distance_inter_area             = 110
  distance_intra_area             = 110
  log_neighbour_changes           = "enable"
  router_id                       = "0.0.0.0"

  area {
    id = "0.0.0.0"
  }

  ospf_interface {
    name      = local.aci_interface
    interface = local.aci_interface
    status    = "enable"
  }

  network {
    # this disgusting monster crafts prefix from aci interface IP
    prefix = join("/", [cidrhost(join("/", [split(" ", lookup(module.fgt_interfaces.interfaces["${local.aci_interface}"], "ip"))[0], lookup(local.netmask2cidr, split(" ", lookup(module.fgt_interfaces.interfaces["${local.aci_interface}"], "ip"))[1])]), 0), lookup(local.netmask2cidr, split(" ", lookup(module.fgt_interfaces.interfaces["${local.aci_interface}"], "ip"))[1])])
  }

  redistribute {
    metric      = 0
    metric_type = "2"
    name        = "connected"
    status      = "disable"
    tag         = 0
  }
  redistribute {
    metric      = 0
    metric_type = "2"
    name        = "static"
    status      = "enable"
    tag         = 0
  }
  redistribute {
    metric      = 0
    metric_type = "2"
    name        = "rip"
    status      = "disable"
    tag         = 0
  }
  redistribute {
    metric      = 0
    metric_type = "2"
    name        = "bgp"
    status      = "disable"
    tag         = 0
  }
  redistribute {
    metric      = 0
    metric_type = "2"
    name        = "isis"
    status      = "disable"
    tag         = 0
  }

}
```
Fortigate config:
```
sogavegur-fw (TEST) # show router ospf
path=router, objname=ospf, tablename=(null), size=296
config router ospf
    set default-information-originate enable
    config area
        edit 0.0.0.0
        next
    end
    config ospf-interface
        edit "TEST-TN-ACI"
            set interface "TEST-TN-ACI"
        next
    end
    config network
        edit 1
            set prefix 10.240.42.0 255.255.255.248
        next
    end
    config redistribute "connected"
    end
    config redistribute "static"
        set status enable
    end
    config redistribute "rip"
    end
    config redistribute "bgp"
    end
    config redistribute "isis"
    end
end

```

Debugs from Fortigate CLI during destroy show no action is attempted during destroy:
```
sogavegur-fw (TEST) # diagnose debug enable
sogavegur-fw (TEST) # diagnose debug application httpsd -1
Debug messages will be on for 30 minutes.
sogavegur-fw (TEST) # diagnose debug cli 8
Debug messages will be on for 30 minutes.
```
```
Terraform will perform the following actions:

  # fortios_router_ospf.ospf will be destroyed
------OMIT-----
fortios_router_ospf.ospf: Destroying... [id=RouterOspf]
fortios_router_ospf.ospf: Destruction complete after 0s

Apply complete! Resources: 0 added, 0 changed, 1 destroyed.
```

Config still exists on Fortigate:
```
sogavegur-fw (TEST) # show router ospf
path=router, objname=ospf, tablename=(null), size=296
config router ospf
    set default-information-originate enable
    config area
        edit 0.0.0.0
        next
    end
    config ospf-interface
        edit "TEST-TN-ACI"
            set interface "TEST-TN-ACI"
        next
    end
    config network
        edit 1
            set prefix 10.240.42.0 255.255.255.248
        next
    end
    config redistribute "connected"
    end
    config redistribute "static"
        set status enable
    end
    config redistribute "rip"
    end
    config redistribute "bgp"
    end
    config redistribute "isis"
    end
end
```

**Comment by frankshen01** [comment]
Hi @poroping , thank you for raising this issue.  It will be fixed in the next release.

**Comment by frankshen01** [comment]
Supported and released, please see the latest version(v1.7.0): https://registry.terraform.io/providers/fortinetdev/fortios/latest 

**Validation:**
```
# cat main.tf
provider "fortios" {
  hostname = "192.168.52.177"
  insecure = "true"
}

resource "fortios_router_ospf" "ospf" {
  default_information_metric      = 10
  default_information_metric_type = "2"
  default_information_originate   = "enable"
  default_metric                  = 10
  distance                        = 110
  distance_external               = 110
  distance_inter_area             = 110
  distance_intra_area             = 110
  log_neighbour_changes           = "enable"
  router_id                       = "0.0.0.0"

  area {
    id = "0.0.0.0"
  }

  network {
    prefix = "10.240.42.0 255.255.255.248"
  }

  ospf_interface {
    name      = "fortilink"
    interface = "fortilink"
    status    = "enable"
  }
}

# terraform apply

mysho (ospf) # show
config router ospf
    set default-information-originate enable
    config area
        edit 0.0.0.0
        next
    end
    config ospf-interface
        edit "fortilink"
            set interface "fortilink"
        next
    end
    config network
        edit 1
            set prefix 10.240.42.0 255.255.255.248
        next
    end
    config redistribute "connected"
    end
    config redistribute "static"
        set status enable
    end
    config redistribute "rip"
    end
    config redistribute "bgp"
    end
    config redistribute "isis"
    end
end

# terraform destroy

mysho (ospf) # show
config router ospf
    set default-information-originate enable
    config redistribute "connected"
    end
    config redistribute "static"
        set status enable
    end
    config redistribute "rip"
    end
    config redistribute "bgp"
    end
    config redistribute "isis"
    end
end

```
Thanks!

**Comment by poroping** [comment]
I can confirm this works as of v1.7.0, thanks.
